cmd/kubelet/app/server.go

```diff
@@ -597,6 +622,94 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
```

Comment:
Not familiar with `CreateAPIServerClientConfig` but `updateTransport` is called where the certificate manager is initialized, so if there are other clientConfigs that aren't getting updated then that should mean they don't have an associated certificate manager. Are they using the same certificates?

In order to get live certificate rotation, I need to attach a callback function to the tls.Transport. Do you have a different approach in mind you would like to suggest?